galsim/phase_psf.py

```diff
@@ -1101,6 +1181,35 @@ def __init__(self, screen_list, lam, t0=0.0, exptime=0.0, time_step=0.025, flux=
 
         self._screen_list._delayCalculation(self)
 
+    @property
+    def kcrit(self):
+        if self._kcrit is None:
+            r0_500 = self._screen_list.r0_500_effective
+            if r0_500 is None:  # No AtmosphericScreens in list
+                self._kcrit = float('inf')
+            else:
+                r0 = r0_500 * (self.lam/500.)**(6./5)
+                self._kcrit = self._kcrit_factor / r0
+        return self._kcrit
+
+    @property
+    def second_kick(self):
+        """Make a SecondKick object based on contents of _screen_list and aper.
+        """
+        if self._second_kick is None:
+            r0_500 = self._screen_list.r0_500_effective
+            if r0_500 is None:  # No AtmosphericScreens in list
+                self._second_kick = galsim.Airy(
+                        lam=self.lam, diam=self.aper.diam, obscuration=self.aper.obscuration)
+            else:
+                L0 = self._screen_list.L0_effective
+                r0 = r0_500 * (self.lam/500.)**(6./5)
+                self._second_kick = galsim.SecondKick(
+                        self.lam, r0, self.aper.diam, self.aper.obscuration,
+                        L0=L0, kcrit=self.kcrit, scale_unit=self.scale_unit,
+                        gsparams=self._gsparams)
+        return self._second_kick
```

Comment:
So I like the `@lazy_property` idea, but I noticed that using this does remove the helpful documentation that appears for a regular `@property` when one does, for example, `help(myPhaseScreenPSF)`.  It would be nice if we could figure out a way for lazy_property to maintain the documentation.